official/transformer/v2/transformer_benchmark.py

```diff
@@ -83,10 +83,22 @@ def _run_and_report_benchmark(self,
 
     metrics = []
     if 'bleu_uncased' in stats:
-      metrics.append({'name': 'bleu_uncased',
-                      'value': stats['bleu_uncased'],
-                      'min_value': bleu_min,
-                      'max_value': bleu_max})
+      if 'bleu_uncased_history' in stats:
+        bleu_uncased_best = max(stats['bleu_uncased_history'],
+                                key=lambda x: x[1])
+        metrics.append({'name': 'bleu_uncased',
+                        'value': bleu_uncased_best[1],
+                        'min_value': bleu_min,
+                        'max_value': bleu_max})
+        metrics.append({'name': 'bleu_recorded_iteration',
```

Comment:
From the name bleu_recorded_iteration it wasn't immediately clear to me that you meant the iteration whether the best score was recorded.  I'd suggest bleu_best_score_iteration.

Reply:
fixed and this is one of the best part of code reviews.  More ideas on naming stuff.  Thank you.